packages/web-client/app/components/card-pay/issue-prepaid-card-workflow/preview/index.ts

```diff
@@ -51,13 +79,30 @@ export default class CardPayPrepaidCardWorkflowPreviewComponent extends Componen
         patternId: workflowSession.state.pattern.id,
       });
 
-      let prepaidCardSafe = yield taskFor(
-        this.layer2Network.issuePrepaidCard
-      ).perform(this.faceValue, customization.did, {
+      this.chinInProgressMessage =
+        'You will receive a confirmation request from the Card Wallet app in a few moments…';
+      let options: IssuePrepaidCardOptions = {
         onTxHash: (txHash: TransactionHash) => {
           this.txHash = txHash;
+          this.chinInProgressMessage =
+            'Waiting for the transaction to be finalized…';
         },
-      });
+      };
+      if (this.lastNonce) {
+        options.nonce = this.lastNonce;
+      } else {
+        options.onNonce = (nonce: string) => {
+          this.lastNonce = nonce;
+        };
+      }
+      let prepaidCardSafeTaskInstance = taskFor(
+        this.layer2Network.issuePrepaidCard
+      ).perform(this.faceValue, customization.did, options);
+      let prepaidCardSafe = yield race([
+        prepaidCardSafeTaskInstance,
+        taskFor(this.timerTask).perform(),
+      ]);
```

Comment:
Is race here just to cancel the timerTask as soon as the `prepaidCardSafeTaskInstance` resolves?

Reply:
Correct.